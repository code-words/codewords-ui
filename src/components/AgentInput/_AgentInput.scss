input {
  height: 34px;
  padding-left: 5px;
  border-radius: 5px;
}