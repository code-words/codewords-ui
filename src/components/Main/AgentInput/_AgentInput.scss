.AgentInput {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(255, 255, 255, 0.242);
  padding: 25px 20px;
  margin: 10px 10px;
  width: 90%;
  box-sizing: border-box;
  border-radius: 10px;
  border: 1px ridge $player-active;

  input, button {
    height: 30px;
    padding-left: 5px;
    margin-right: 20px;
    border-radius: 5px;
    box-sizing: border-box;
  }

  .send-btn {
    padding: 5px 25px;
    font-size: 18px;
    color: black;
    border: none;
    border-radius: 50px;
    text-transform: uppercase;

    &:disabled {
      background: gray;
      color: rgb(91, 91, 91);

      &:hover {
        background: gray;
      }
    }

    &:hover {
      background: $player-active;
    }
  }
  
  .num-input {
    width: 38px;
  }
  
  .hint-entries {
    display: flex;
    align-items: center;
    justify-content: center;
  
    label {
      padding-right: 5px;
      font-size: 28px;
    }
  }
}