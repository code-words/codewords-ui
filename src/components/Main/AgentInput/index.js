import React, { Component } from 'react';
import ConversationsList from '../../ConversationsList';

class AgentInput extends Component {
  constructor() {
    super()
    this.state = {
      hint: '',
      relWords: 0
    }
  }

  handleChange = (e) => {
    const {name, value} = e.target;

    name === 'hint'
      ? this.setState({ hint: value })
      : this.setState({ relWords: value })
  }

  handleSubmit = (e) => {
    e.preventDefault();
  }

  render() {
    return (
      <form className="AgentInput">
        <ConversationsList name="hint" type="text" value={this.state.hint} onChange={this.handleChange} />
        {/* <input name="hint" type="text" value={this.state.hint} onChange={this.handleChange} /> */}
        <input className="num-input" name="relWords" type="number" value={this.state.relWords} onChange={this.handleChange} />
        {/* <input type="submit" onClick={this.handleSubmit} /> */}
      </form>
    )
  }
}

export default AgentInput;