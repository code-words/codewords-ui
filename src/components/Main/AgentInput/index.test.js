import React from 'react';
import AgentInput from './index';
import { shallow } from 'enzyme';

describe('AgentInput', () => {
  let wrapper;
  let mockFn= {sendHint: jest.fn()}

  beforeEach(() => {
    wrapper = shallow(<AgentInput  cable={mockFn}/>);

  })

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should call handleChange on input', () => {
     let spy = jest.spyOn(wrapper.instance(), "handleChange");
     wrapper.find('.chat-input').simulate("change", {target:{name: "test", value: "testing"}});
    expect(spy).toHaveBeenCalled()
  })
  it('should call handleChatInputKeyPress on keypress', () => {
    let spy = jest.spyOn(wrapper.instance(), "handleChatInputKeyPress");
    wrapper.find('.chat-input').simulate("keyPress", {key: "Enter", preventDefault: jest.fn()});
   expect(spy).toHaveBeenCalled()
 })
 it('should update state when handleSendEvent is called', () => {
  wrapper.setState({ hintWord: 'testing' });
  wrapper.instance().handleSendEvent({preventDefault: jest.fn()})
  expect(wrapper.state('hintWord')).toEqual("");
 });
 it('should call handleSendEvent when form is submitted', () => {
  let spy = jest.spyOn(wrapper.instance(), "handleSendEvent");
  wrapper.find('form').simulate("submit", {key: "Enter", preventDefault: jest.fn()});
   expect(spy).toHaveBeenCalled()
 });
 it('should call handleSendEvent when button is clicked', () => {
  let spy = jest.spyOn(wrapper.instance(), "handleSendEvent");
  wrapper.find('.send-btn').simulate("click", {target:{name: "test", value: "testing"}, preventDefault: jest.fn()});
  expect(spy).toHaveBeenCalled()
 });
  
})