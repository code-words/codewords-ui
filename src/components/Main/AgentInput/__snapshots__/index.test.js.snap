// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentInput should match component snapshot 1`] = `ShallowWrapper {}`;
